### PR TITLE
headers: fix blockheader tests

### PR DESCRIPTION
**Pull request by tynes**
Fixes up the `BlockHeader` and tests for proper serialization and pow hash + other things

### CODE REVIEW

**Comment by tynes** [comment]
got bored so i pushed some more things

File: primitives/src/transaction/transaction.rs [hunk 1]
```diff
@@ -20,7 +20,6 @@ pub struct Transaction {
 impl Transaction {
     pub fn new(locktime: u32, inputs: Vec<Input>, outputs: Vec<Output>) -> Self {
         Transaction {
-            //@todo not sure what exactly what we should be doing here.
```
Comment (tynes) [inline primitives/src/transaction/transaction.rs:23]: The network is on v0 transactions right now so its ok to default them to 0

File: primitives/src/transaction/transaction.rs [hunk 2]
```diff
@@ -20,7 +20,6 @@ pub struct Transaction {
 impl Transaction {
     pub fn new(locktime: u32, inputs: Vec<Input>, outputs: Vec<Output>) -> Self {
```
Comment (tynes) [inline primitives/src/transaction/transaction.rs:21]: I think the locktime should be the final argument

Comment (tynes) [reply primitives/src/transaction/transaction.rs:21]: Unless you did it this way for memory layout reasons? Not a big deal either way

Comment (kilpatty) [reply primitives/src/transaction/transaction.rs:21]: I don't think there was a particular reason here to be honest, so I'll note to change this

Comment (kilpatty) [reply primitives/src/transaction/transaction.rs:21]: I don't think there was a particular reason here to be honest, so I'll note to change this

**Comment by kilpatty** [comment]
LGTM thanks man! I'm planning to get CI up today so future PRs will run tests and whatnot to make things a bit smoother.

**Pull request merged**